openbrews/recipe/detail-edit/EditRecipeCtrl.js

```diff
@@ -126,10 +131,138 @@
         $state.go("recipes");
       };
 
+
+      ////////////////////////////////////////////////////////////
+      // Functions that watch variables and update calculated
+      // values appropriately
+      ///////////////////////////////////////////////////////////
+
+      function updateOG () {
+        $scope.recipe.og = RecipeUtils.calcOG($scope.recipe);
+      }
+
+      function updateFG () {
+        $scope.recipe.fg = RecipeUtils.calcFG($scope.recipe);
+      }
+
+      function updateABV () {
+        $scope.recipe.abv = RecipeUtils.calcABV($scope.recipe);
+      }
+
+      function updateABW () {
+        $scope.recipe.abw = RecipeUtils.calcABW($scope.recipe);
+      }
+
+      function updateIBU() {
+        $scope.recipe.ibu = RecipeUtils.calcIBU($scope.recipe);
+      }
+
+      function updateSRM() {
+        $scope.recipe.srm = RecipeUtils.calcSRM($scope.recipe);
+      }
+
+      function srmWatcher() {
+        updateSRM();
+      }
+
+      function ibuWatcher(newVals, oldVals) {
+        updateIBU();
+      }
+
+      function abwWatcher (newVals, oldVals) {
+        updateABW();
+      }
+
+      function abvWatcher (newVals, oldVals) {
+        updateABW();
+      }
+
+      function fgWatcher (newVals, oldVals) {
+        updateFG();
+        updateABV();
+        updateABW();
+      }
+
+      function ogWatcher (newVals, oldVals) {
+        updateOG();
+        updateFG();
+        updateABV();
+        updateABW();
+        updateIBU();
+      }
+
+      /* Make OG recalculate when it's dependencies change */
+      $scope.$watchGroup(
+        [
+          'recipe.boilSize',
+          'recipe.boilSizeUnits',
+          'recipe.mashEfficiency',
+          'recipe.steepEfficiency'
+        ],
+        ogWatcher);
+
+      $scope.$watchCollection(
+        'recipe.fermentables',
+        function (newVals, oldVals) {
+          var i;
+          for(i = 0; i < newVals.length; ++i) {
+            $scope.$watchCollection('recipe.fermentables[' + i + ']', ogWatcher);
```

Comment:
@mdw7326 I think that only one `$scope.$watchCollection` is required. Inside this loop, you can also add `srmWatcher`. And similarly, in the yeasts, you can add both `fgWatcher` and `abvWatcher`. I know that this code was written by me, but I just realised that multiple `$watchCollection`s are not required.